test/e2e/integration_test.go

```diff
@@ -0,0 +1,116 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+
+	// bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"net/http"
+	// "k8s.io/utils/pointer"
+
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	// "sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	// "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	k8sVersion          string
+	containerRuntime string
+	err                 error
+)
+
+var _ = Describe("When testing integration [integration]", func() {
+	BeforeEach(func() {
```

Comment:
Please not that we select the cluster template based on `osType`  https://github.com/metal3-io/cluster-api-provider-metal3/blob/3c299790d3af391e85de6b64852c2fd5ae0a953c/test/e2e/pivoting_based_feature_test.go#L31 
that is used in `createTargetCluster` function 
https://github.com/metal3-io/cluster-api-provider-metal3/blob/3c299790d3af391e85de6b64852c2fd5ae0a953c/test/e2e/pivoting_based_feature_test.go#L134